shader error with programmable renderer in AreaLightExample on Windows / VS

It seems like it doesn't like the:
#define MAX_LIGHTS 
#define USE_TEXTURE

The shader doesn't compile, but the app runs. 
Going to paste the print out.

```
[ error ] ofShader: setupShaderFromSource(): GL_FRAGMENT_SHADER shader failed to
 compile
[ error ] ofShader: GL_FRAGMENT_SHADER shader reports:
0(10) : error C0000: syntax error, unexpected $undefined at token "#"
0(10) : error C0000: syntax error, unexpected $undefined at token "#"
0(10) : error C0000: syntax error, unexpected $undefined at token "#"

[ error ] ofShader: GL_FRAGMENT_SHADER, offending line 10 :
            8   #define SAMPLER sampler2D
            9   #define MAX_LIGHTS 1
           10   IN vec2 outtexcoord; IN vec3 transformedNormal; IN vec3 eyePosit
ion3; struct lightData { float enabled; vec4 ambient; float type; vec4 position;
 vec4 diffuse; vec4 specular; float constantAttenuation; float linearAttenuation
; float quadraticAttenuation; float spotCutoff; float spotCosCutoff; float spotE
xponent; vec3 spotDirection; vec3 halfVector; float width; float height; vec3 ri
ght; vec3 up; }; uniform SAMPLER tex0; uniform vec4 mat_ambient; uniform vec4 ma
t_diffuse; uniform vec4 mat_specular; uniform vec4 mat_emissive; uniform float m
at_shininess; uniform vec4 global_ambient; uniform mat4 modelViewMatrix; uniform
 mat4 projectionMatrix; uniform mat4 textureMatrix; uniform mat4 modelViewProjec
tionMatrix; uniform lightData lights[MAX_LIGHTS]; void pointLight( in lightData
light, in vec3 normal, in vec3 ecPosition3, inout vec3 ambient, inout vec3 diffu
se, inout vec3 specular ){ float nDotVP; float nDotHV; float pf; float attenuati
on; float d; vec3 VP; vec3 halfVector; vec3 eye = vec3 (0.0, 0.0, 1.0); VP = vec
3 (light.position.xyz) - ecPosition3; d = length(VP); attenuation = 1.0 / (light
.constantAttenuation + light.linearAttenuation * d + light.quadraticAttenuation
* d * d); VP = normalize(VP); halfVector = normalize(VP + eye); nDotHV = max(0.0
, dot(normal, halfVector)); nDotVP = max(0.0, dot(normal, VP)); ambient += light
.ambient.rgb * attenuation; diffuse += light.diffuse.rgb * nDotVP * attenuation;
 #ifndef TARGET_OPENGLES #define SPECULAR_REFLECTION #endif #ifndef SPECULAR_REF
LECTION pf = mix(0.0, pow(nDotHV, mat_shininess), step(0.0000001, nDotVP)); spec
ular += light.specular.rgb * pf * nDotVP * attenuation; #else float w = pow(1.0
- max(0.0, dot(halfVector, VP)), 5.0); vec3 specularReflection = attenuation * v
ec3(light.specular.rgb) * mix(vec3(mat_specular.rgb), vec3(1.0), w) * pow(nDotHV
, mat_shininess); specular += mix(vec3(0.0), specularReflection, step(0.0000001,
 nDotVP)); #endif } void directionalLight(in lightData light, in vec3 normal, in
out vec3 ambient, inout vec3 diffuse, inout vec3 specular){ float nDotVP; float
nDotHV; float pf; nDotVP = max(0.0, dot(normal, normalize(vec3(light.position)))
); nDotHV = max(0.0, dot(normal, light.halfVector)); pf = mix(0.0, pow(nDotHV, m
at_shininess), step(0.0000001, nDotVP)); ambient += light.ambient.rgb; diffuse +
= light.diffuse.rgb * nDotVP; specular += light.specular.rgb * pf * nDotVP; } vo
id spotLight(in lightData light, in vec3 normal, in vec3 ecPosition3, inout vec3
 ambient, inout vec3 diffuse, inout vec3 specular){ float nDotVP; float nDotHV;
float pf; float d; vec3 VP; vec3 eye = vec3 (0.0, 0.0, 1.0); float spotEffect; f
loat attenuation=1.0; vec3 halfVector; VP = light.position.xyz - ecPosition3; sp
otEffect = dot(light.spotDirection, -normalize(VP)); if (spotEffect > light.spot
CosCutoff) { d = length(VP); spotEffect = pow(spotEffect, light.spotExponent); a
ttenuation = spotEffect / (light.constantAttenuation + light.linearAttenuation *
 d + light.quadraticAttenuation * d * d); VP = normalize(VP); halfVector = norma
lize(VP + eye); nDotHV = max(0.0, dot(normal, halfVector)); nDotVP = max(0.0, do
t(normal, VP)); pf = mix(0.0, pow(nDotHV, mat_shininess), step(0.0000001, nDotVP
)); diffuse += light.diffuse.rgb * nDotVP * attenuation; specular += light.specu
lar.rgb * pf * nDotVP * attenuation; } ambient += light.ambient.rgb * attenuatio
n; } vec3 projectOnPlane(in vec3 point, in vec3 planeCenter, in vec3 planeNormal
){ return point - dot( point - planeCenter, planeNormal ) * planeNormal; } vec3
linePlaneIntersect(in vec3 lp, in vec3 lv, in vec3 pc, in vec3 pn){ return lp+lv
*(dot(pn,pc-lp)/dot(pn,lv)); } void areaLight(in lightData light, in vec3 N, in
vec3 V, inout vec3 ambient, inout vec3 diffuse, inout vec3 specular){ vec3 right
 = light.right; vec3 pnormal = light.spotDirection; vec3 up = light.up; float wi
dth = light.width*0.5; float height = light.height*0.5; float attenuation; vec3
projection = projectOnPlane(V,light.position.xyz,pnormal); vec3 dir = projection
-light.position.xyz; vec2 diagonal = vec2(dot(dir,right),dot(dir,up)); vec2 near
est2D = vec2(clamp( diagonal.x,-width,width ),clamp( diagonal.y,-height,height))
; vec3 nearestPointInside = vec3(light.position.xyz)+(right*nearest2D.x+up*neare
st2D.y); float dist = distance(V,nearestPointInside); vec3 lightDir = normalize(
nearestPointInside - V); attenuation = 1.0 / (light.constantAttenuation + light.
linearAttenuation * dist + light.quadraticAttenuation * dist * dist); float Ndot
L = max( dot( pnormal, -lightDir ), 0.0 ); float NdotL2 = max( dot( N, lightDir
), 0.0 ); if ( NdotL * NdotL2 > 0.0 ) { float diffuseFactor = sqrt( NdotL * Ndot
L2 ); vec3 R = reflect( normalize( -V ), N ); vec3 E = linePlaneIntersect( V, R,
 light.position.xyz, pnormal ); float specAngle = dot( R, pnormal ); if (specAng
le > 0.0){ vec3 dirSpec = E - light.position.xyz; vec2 dirSpec2D = vec2(dot(dirS
pec,right),dot(dirSpec,up)); vec2 nearestSpec2D = vec2(clamp( dirSpec2D.x,-width
,width ),clamp( dirSpec2D.y,-height,height)); float specFactor = 1.0-clamp(lengt
h(nearestSpec2D-dirSpec2D) * 0.05 * mat_shininess,0.0,1.0); specular += light.sp
ecular.rgb * specFactor * specAngle * diffuseFactor * attenuation; } diffuse +=
light.diffuse.rgb * diffuseFactor * attenuation; } ambient += light.ambient.rgb
* attenuation; } void main (void){ vec3 ambient = global_ambient.rgb; vec3 diffu
se = vec3(0.0,0.0,0.0); vec3 specular = vec3(0.0,0.0,0.0); for( int i = 0; i < M
AX_LIGHTS; i++ ){ if(lights[i].enabled<0.5) continue; if(lights[i].type<0.5){ po
intLight(lights[i], transformedNormal, eyePosition3, ambient, diffuse, specular)
; }else if(lights[i].type<1.5){ directionalLight(lights[i], transformedNormal, a
mbient, diffuse, specular); }else if(lights[i].type<2.5){ spotLight(lights[i], t
ransformedNormal, eyePosition3, ambient, diffuse, specular); }else{ areaLight(li
ghts[i], transformedNormal, eyePosition3, ambient, diffuse, specular); } }
           11   #ifdef HAS_TEXTURE
           12    vec4 tex = TEXTURE(tex0, outtexcoord); vec4 localColor = vec4(a
mbient,1.0) * mat_ambient + vec4(diffuse,1.0) * tex + vec4(specular,1.0) * mat_s
pecular + mat_emissive;

[ error ] ofShader: checkProgramLinkStatus(): program failed to link
[ error ] ofShader: ofShader: program reports:
Fragment info
-------------
0(10) : error C0000: syntax error, unexpected $undefined at token "#"
0(10) : error C0000: syntax error, unexpected $undefined at token "#"
0(10) : error C0000: syntax error, unexpected $undefined at token "#"

[ error ] ofShader: setupShaderFromSource(): GL_FRAGMENT_SHADER shader failed to
 compile
[ error ] ofShader: GL_FRAGMENT_SHADER shader reports:
0(11) : error C0000: syntax error, unexpected $undefined at token "#"
0(11) : error C0000: syntax error, unexpected $undefined at token "#"
0(11) : error C0000: syntax error, unexpected $undefined at token "#"

[ error ] ofShader: GL_FRAGMENT_SHADER, offending line 11 :
            9   #define MAX_LIGHTS 1
           10   #define HAS_TEXTURE
           11   IN vec2 outtexcoord; IN vec3 transformedNormal; IN vec3 eyePosit
ion3; struct lightData { float enabled; vec4 ambient; float type; vec4 position;
 vec4 diffuse; vec4 specular; float constantAttenuation; float linearAttenuation
; float quadraticAttenuation; float spotCutoff; float spotCosCutoff; float spotE
xponent; vec3 spotDirection; vec3 halfVector; float width; float height; vec3 ri
ght; vec3 up; }; uniform SAMPLER tex0; uniform vec4 mat_ambient; uniform vec4 ma
t_diffuse; uniform vec4 mat_specular; uniform vec4 mat_emissive; uniform float m
at_shininess; uniform vec4 global_ambient; uniform mat4 modelViewMatrix; uniform
 mat4 projectionMatrix; uniform mat4 textureMatrix; uniform mat4 modelViewProjec
tionMatrix; uniform lightData lights[MAX_LIGHTS]; void pointLight( in lightData
light, in vec3 normal, in vec3 ecPosition3, inout vec3 ambient, inout vec3 diffu
se, inout vec3 specular ){ float nDotVP; float nDotHV; float pf; float attenuati
on; float d; vec3 VP; vec3 halfVector; vec3 eye = vec3 (0.0, 0.0, 1.0); VP = vec
3 (light.position.xyz) - ecPosition3; d = length(VP); attenuation = 1.0 / (light
.constantAttenuation + light.linearAttenuation * d + light.quadraticAttenuation
* d * d); VP = normalize(VP); halfVector = normalize(VP + eye); nDotHV = max(0.0
, dot(normal, halfVector)); nDotVP = max(0.0, dot(normal, VP)); ambient += light
.ambient.rgb * attenuation; diffuse += light.diffuse.rgb * nDotVP * attenuation;
 #ifndef TARGET_OPENGLES #define SPECULAR_REFLECTION #endif #ifndef SPECULAR_REF
LECTION pf = mix(0.0, pow(nDotHV, mat_shininess), step(0.0000001, nDotVP)); spec
ular += light.specular.rgb * pf * nDotVP * attenuation; #else float w = pow(1.0
- max(0.0, dot(halfVector, VP)), 5.0); vec3 specularReflection = attenuation * v
ec3(light.specular.rgb) * mix(vec3(mat_specular.rgb), vec3(1.0), w) * pow(nDotHV
, mat_shininess); specular += mix(vec3(0.0), specularReflection, step(0.0000001,
 nDotVP)); #endif } void directionalLight(in lightData light, in vec3 normal, in
out vec3 ambient, inout vec3 diffuse, inout vec3 specular){ float nDotVP; float
nDotHV; float pf; nDotVP = max(0.0, dot(normal, normalize(vec3(light.position)))
); nDotHV = max(0.0, dot(normal, light.halfVector)); pf = mix(0.0, pow(nDotHV, m
at_shininess), step(0.0000001, nDotVP)); ambient += light.ambient.rgb; diffuse +
= light.diffuse.rgb * nDotVP; specular += light.specular.rgb * pf * nDotVP; } vo
id spotLight(in lightData light, in vec3 normal, in vec3 ecPosition3, inout vec3
 ambient, inout vec3 diffuse, inout vec3 specular){ float nDotVP; float nDotHV;
float pf; float d; vec3 VP; vec3 eye = vec3 (0.0, 0.0, 1.0); float spotEffect; f
loat attenuation=1.0; vec3 halfVector; VP = light.position.xyz - ecPosition3; sp
otEffect = dot(light.spotDirection, -normalize(VP)); if (spotEffect > light.spot
CosCutoff) { d = length(VP); spotEffect = pow(spotEffect, light.spotExponent); a
ttenuation = spotEffect / (light.constantAttenuation + light.linearAttenuation *
 d + light.quadraticAttenuation * d * d); VP = normalize(VP); halfVector = norma
lize(VP + eye); nDotHV = max(0.0, dot(normal, halfVector)); nDotVP = max(0.0, do
t(normal, VP)); pf = mix(0.0, pow(nDotHV, mat_shininess), step(0.0000001, nDotVP
)); diffuse += light.diffuse.rgb * nDotVP * attenuation; specular += light.specu
lar.rgb * pf * nDotVP * attenuation; } ambient += light.ambient.rgb * attenuatio
n; } vec3 projectOnPlane(in vec3 point, in vec3 planeCenter, in vec3 planeNormal
){ return point - dot( point - planeCenter, planeNormal ) * planeNormal; } vec3
linePlaneIntersect(in vec3 lp, in vec3 lv, in vec3 pc, in vec3 pn){ return lp+lv
*(dot(pn,pc-lp)/dot(pn,lv)); } void areaLight(in lightData light, in vec3 N, in
vec3 V, inout vec3 ambient, inout vec3 diffuse, inout vec3 specular){ vec3 right
 = light.right; vec3 pnormal = light.spotDirection; vec3 up = light.up; float wi
dth = light.width*0.5; float height = light.height*0.5; float attenuation; vec3
projection = projectOnPlane(V,light.position.xyz,pnormal); vec3 dir = projection
-light.position.xyz; vec2 diagonal = vec2(dot(dir,right),dot(dir,up)); vec2 near
est2D = vec2(clamp( diagonal.x,-width,width ),clamp( diagonal.y,-height,height))
; vec3 nearestPointInside = vec3(light.position.xyz)+(right*nearest2D.x+up*neare
st2D.y); float dist = distance(V,nearestPointInside); vec3 lightDir = normalize(
nearestPointInside - V); attenuation = 1.0 / (light.constantAttenuation + light.
linearAttenuation * dist + light.quadraticAttenuation * dist * dist); float Ndot
L = max( dot( pnormal, -lightDir ), 0.0 ); float NdotL2 = max( dot( N, lightDir
), 0.0 ); if ( NdotL * NdotL2 > 0.0 ) { float diffuseFactor = sqrt( NdotL * Ndot
L2 ); vec3 R = reflect( normalize( -V ), N ); vec3 E = linePlaneIntersect( V, R,
 light.position.xyz, pnormal ); float specAngle = dot( R, pnormal ); if (specAng
le > 0.0){ vec3 dirSpec = E - light.position.xyz; vec2 dirSpec2D = vec2(dot(dirS
pec,right),dot(dirSpec,up)); vec2 nearestSpec2D = vec2(clamp( dirSpec2D.x,-width
,width ),clamp( dirSpec2D.y,-height,height)); float specFactor = 1.0-clamp(lengt
h(nearestSpec2D-dirSpec2D) * 0.05 * mat_shininess,0.0,1.0); specular += light.sp
ecular.rgb * specFactor * specAngle * diffuseFactor * attenuation; } diffuse +=
light.diffuse.rgb * diffuseFactor * attenuation; } ambient += light.ambient.rgb
* attenuation; } void main (void){ vec3 ambient = global_ambient.rgb; vec3 diffu
se = vec3(0.0,0.0,0.0); vec3 specular = vec3(0.0,0.0,0.0); for( int i = 0; i < M
AX_LIGHTS; i++ ){ if(lights[i].enabled<0.5) continue; if(lights[i].type<0.5){ po
intLight(lights[i], transformedNormal, eyePosition3, ambient, diffuse, specular)
; }else if(lights[i].type<1.5){ directionalLight(lights[i], transformedNormal, a
mbient, diffuse, specular); }else if(lights[i].type<2.5){ spotLight(lights[i], t
ransformedNormal, eyePosition3, ambient, diffuse, specular); }else{ areaLight(li
ghts[i], transformedNormal, eyePosition3, ambient, diffuse, specular); } }
           12   #ifdef HAS_TEXTURE
           13    vec4 tex = TEXTURE(tex0, outtexcoord); vec4 localColor = vec4(a
mbient,1.0) * mat_ambient + vec4(diffuse,1.0) * tex + vec4(specular,1.0) * mat_s
pecular + mat_emissive;

[ error ] ofShader: checkProgramLinkStatus(): program failed to link
[ error ] ofShader: ofShader: program reports:
Fragment info
-------------
0(11) : error C0000: syntax error, unexpected $undefined at token "#"
0(11) : error C0000: syntax error, unexpected $undefined at token "#"
0(11) : error C0000: syntax error, unexpected $undefined at token "#"

[ error ] ofShader: setupShaderFromSource(): GL_FRAGMENT_SHADER shader failed to
 compile
[ error ] ofShader: GL_FRAGMENT_SHADER shader reports:
0(11) : error C0000: syntax error, unexpected $undefined at token "#"
0(11) : error C0000: syntax error, unexpected $undefined at token "#"
0(11) : error C0000: syntax error, unexpected $undefined at token "#"

[ error ] ofShader: GL_FRAGMENT_SHADER, offending line 11 :
            9   #define MAX_LIGHTS 1
           10   #define HAS_TEXTURE
           11   IN vec2 outtexcoord; IN vec3 transformedNormal; IN vec3 eyePosit
ion3; struct lightData { float enabled; vec4 ambient; float type; vec4 position;
 vec4 diffuse; vec4 specular; float constantAttenuation; float linearAttenuation
; float quadraticAttenuation; float spotCutoff; float spotCosCutoff; float spotE
xponent; vec3 spotDirection; vec3 halfVector; float width; float height; vec3 ri
ght; vec3 up; }; uniform SAMPLER tex0; uniform vec4 mat_ambient; uniform vec4 ma
t_diffuse; uniform vec4 mat_specular; uniform vec4 mat_emissive; uniform float m
at_shininess; uniform vec4 global_ambient; uniform mat4 modelViewMatrix; uniform
 mat4 projectionMatrix; uniform mat4 textureMatrix; uniform mat4 modelViewProjec
tionMatrix; uniform lightData lights[MAX_LIGHTS]; void pointLight( in lightData
light, in vec3 normal, in vec3 ecPosition3, inout vec3 ambient, inout vec3 diffu
se, inout vec3 specular ){ float nDotVP; float nDotHV; float pf; float attenuati
on; float d; vec3 VP; vec3 halfVector; vec3 eye = vec3 (0.0, 0.0, 1.0); VP = vec
3 (light.position.xyz) - ecPosition3; d = length(VP); attenuation = 1.0 / (light
.constantAttenuation + light.linearAttenuation * d + light.quadraticAttenuation
* d * d); VP = normalize(VP); halfVector = normalize(VP + eye); nDotHV = max(0.0
, dot(normal, halfVector)); nDotVP = max(0.0, dot(normal, VP)); ambient += light
.ambient.rgb * attenuation; diffuse += light.diffuse.rgb * nDotVP * attenuation;
 #ifndef TARGET_OPENGLES #define SPECULAR_REFLECTION #endif #ifndef SPECULAR_REF
LECTION pf = mix(0.0, pow(nDotHV, mat_shininess), step(0.0000001, nDotVP)); spec
ular += light.specular.rgb * pf * nDotVP * attenuation; #else float w = pow(1.0
- max(0.0, dot(halfVector, VP)), 5.0); vec3 specularReflection = attenuation * v
ec3(light.specular.rgb) * mix(vec3(mat_specular.rgb), vec3(1.0), w) * pow(nDotHV
, mat_shininess); specular += mix(vec3(0.0), specularReflection, step(0.0000001,
 nDotVP)); #endif } void directionalLight(in lightData light, in vec3 normal, in
out vec3 ambient, inout vec3 diffuse, inout vec3 specular){ float nDotVP; float
nDotHV; float pf; nDotVP = max(0.0, dot(normal, normalize(vec3(light.position)))
); nDotHV = max(0.0, dot(normal, light.halfVector)); pf = mix(0.0, pow(nDotHV, m
at_shininess), step(0.0000001, nDotVP)); ambient += light.ambient.rgb; diffuse +
= light.diffuse.rgb * nDotVP; specular += light.specular.rgb * pf * nDotVP; } vo
id spotLight(in lightData light, in vec3 normal, in vec3 ecPosition3, inout vec3
 ambient, inout vec3 diffuse, inout vec3 specular){ float nDotVP; float nDotHV;
float pf; float d; vec3 VP; vec3 eye = vec3 (0.0, 0.0, 1.0); float spotEffect; f
loat attenuation=1.0; vec3 halfVector; VP = light.position.xyz - ecPosition3; sp
otEffect = dot(light.spotDirection, -normalize(VP)); if (spotEffect > light.spot
CosCutoff) { d = length(VP); spotEffect = pow(spotEffect, light.spotExponent); a
ttenuation = spotEffect / (light.constantAttenuation + light.linearAttenuation *
 d + light.quadraticAttenuation * d * d); VP = normalize(VP); halfVector = norma
lize(VP + eye); nDotHV = max(0.0, dot(normal, halfVector)); nDotVP = max(0.0, do
t(normal, VP)); pf = mix(0.0, pow(nDotHV, mat_shininess), step(0.0000001, nDotVP
)); diffuse += light.diffuse.rgb * nDotVP * attenuation; specular += light.specu
lar.rgb * pf * nDotVP * attenuation; } ambient += light.ambient.rgb * attenuatio
n; } vec3 projectOnPlane(in vec3 point, in vec3 planeCenter, in vec3 planeNormal
){ return point - dot( point - planeCenter, planeNormal ) * planeNormal; } vec3
linePlaneIntersect(in vec3 lp, in vec3 lv, in vec3 pc, in vec3 pn){ return lp+lv
*(dot(pn,pc-lp)/dot(pn,lv)); } void areaLight(in lightData light, in vec3 N, in
vec3 V, inout vec3 ambient, inout vec3 diffuse, inout vec3 specular){ vec3 right
 = light.right; vec3 pnormal = light.spotDirection; vec3 up = light.up; float wi
dth = light.width*0.5; float height = light.height*0.5; float attenuation; vec3
projection = projectOnPlane(V,light.position.xyz,pnormal); vec3 dir = projection
-light.position.xyz; vec2 diagonal = vec2(dot(dir,right),dot(dir,up)); vec2 near
est2D = vec2(clamp( diagonal.x,-width,width ),clamp( diagonal.y,-height,height))
; vec3 nearestPointInside = vec3(light.position.xyz)+(right*nearest2D.x+up*neare
st2D.y); float dist = distance(V,nearestPointInside); vec3 lightDir = normalize(
nearestPointInside - V); attenuation = 1.0 / (light.constantAttenuation + light.
linearAttenuation * dist + light.quadraticAttenuation * dist * dist); float Ndot
L = max( dot( pnormal, -lightDir ), 0.0 ); float NdotL2 = max( dot( N, lightDir
), 0.0 ); if ( NdotL * NdotL2 > 0.0 ) { float diffuseFactor = sqrt( NdotL * Ndot
L2 ); vec3 R = reflect( normalize( -V ), N ); vec3 E = linePlaneIntersect( V, R,
 light.position.xyz, pnormal ); float specAngle = dot( R, pnormal ); if (specAng
le > 0.0){ vec3 dirSpec = E - light.position.xyz; vec2 dirSpec2D = vec2(dot(dirS
pec,right),dot(dirSpec,up)); vec2 nearestSpec2D = vec2(clamp( dirSpec2D.x,-width
,width ),clamp( dirSpec2D.y,-height,height)); float specFactor = 1.0-clamp(lengt
h(nearestSpec2D-dirSpec2D) * 0.05 * mat_shininess,0.0,1.0); specular += light.sp
ecular.rgb * specFactor * specAngle * diffuseFactor * attenuation; } diffuse +=
light.diffuse.rgb * diffuseFactor * attenuation; } ambient += light.ambient.rgb
* attenuation; } void main (void){ vec3 ambient = global_ambient.rgb; vec3 diffu
se = vec3(0.0,0.0,0.0); vec3 specular = vec3(0.0,0.0,0.0); for( int i = 0; i < M
AX_LIGHTS; i++ ){ if(lights[i].enabled<0.5) continue; if(lights[i].type<0.5){ po
intLight(lights[i], transformedNormal, eyePosition3, ambient, diffuse, specular)
; }else if(lights[i].type<1.5){ directionalLight(lights[i], transformedNormal, a
mbient, diffuse, specular); }else if(lights[i].type<2.5){ spotLight(lights[i], t
ransformedNormal, eyePosition3, ambient, diffuse, specular); }else{ areaLight(li
ghts[i], transformedNormal, eyePosition3, ambient, diffuse, specular); } }
           12   #ifdef HAS_TEXTURE
           13    vec4 tex = TEXTURE(tex0, outtexcoord); vec4 localColor = vec4(a
mbient,1.0) * mat_ambient + vec4(diffuse,1.0) * tex + vec4(specular,1.0) * mat_s
pecular + mat_emissive;

[ error ] ofShader: checkProgramLinkStatus(): program failed to link
[ error ] ofShader: ofShader: program reports:
Fragment info
-------------
0(11) : error C0000: syntax error, unexpected $undefined at token "#"
0(11) : error C0000: syntax error, unexpected $undefined at token "#"
0(11) : error C0000: syntax error, unexpected $undefined at token "#"
```

--- a/libs/openFrameworks/gl/ofMaterial.cpp
+++ b/libs/openFrameworks/gl/ofMaterial.cpp
@@ -308,13 +308,9 @@ static const string fragmentShader = R"(
 
 		ambient += light.ambient.rgb * attenuation;
 		diffuse += light.diffuse.rgb * nDotVP * attenuation;
-)"
 #ifndef TARGET_OPENGLES
-	R"(
 #define SPECULAR_REFLECTION
-	)"
 #endif
-R"(
 #ifndef SPECULAR_REFLECTION
 		// ha! no branching :)
 		pf = mix(0.0, pow(nDotHV, mat_shininess), step(0.0000001, nDotVP));

--- a/libs/openFrameworks/gl/ofMaterial.cpp
+++ b/libs/openFrameworks/gl/ofMaterial.cpp
@@ -197,9 +197,7 @@ void ofMaterial::updateLights(const ofShader & shader,ofGLProgrammableRenderer &
 	}
 }
 
-#define STRINGIFY(x) #x
-
-static const string vertexShader = STRINGIFY(
+static const string vertexShader = R"(
 	OUT vec2 outtexcoord; // pass the texCoord if needed
 	OUT vec3 transformedNormal;
 	OUT vec3 eyePosition3;
@@ -226,10 +224,10 @@ static const string vertexShader = STRINGIFY(
 		outtexcoord = (textureMatrix*vec4(texcoord.x,texcoord.y,0,1)).xy;
 		gl_Position = modelViewProjectionMatrix * position;
 	}
-);
+)";
 
 
-static const string fragmentShader = STRINGIFY(
+static const string fragmentShader = R"(
 	IN vec2 outtexcoord; // pass the texCoord if needed
 	IN vec3 transformedNormal;
 	// Eye-coordinate position of vertex
@@ -310,9 +308,13 @@ static const string fragmentShader = STRINGIFY(
 
 		ambient += light.ambient.rgb * attenuation;
 		diffuse += light.diffuse.rgb * nDotVP * attenuation;
+)"
 #ifndef TARGET_OPENGLES
+	R"(
 #define SPECULAR_REFLECTION
+	)"
 #endif
+R"(
 #ifndef SPECULAR_REFLECTION
 		// ha! no branching :)
 		pf = mix(0.0, pow(nDotHV, mat_shininess), step(0.0000001, nDotVP));
@@ -459,15 +461,15 @@ static const string fragmentShader = STRINGIFY(
 
 		////////////////////////////////////////////////////////////
 		// now add the material info
-		\n#ifdef HAS_TEXTURE\n
+		#ifdef HAS_TEXTURE
 			vec4 tex = TEXTURE(tex0, outtexcoord);
 			vec4 localColor = vec4(ambient,1.0) * mat_ambient + vec4(diffuse,1.0) * tex + vec4(specular,1.0) * mat_specular + mat_emissive;
-		\n#else\n
+		#else
 			vec4 localColor = vec4(ambient,1.0) * mat_ambient + vec4(diffuse,1.0) * mat_diffuse + vec4(specular,1.0) * mat_specular + mat_emissive;
-		\n#endif\n
+		#endif
 		FRAG_COLOR = clamp( localColor, 0.0, 1.0 );
 	}
-);
+)";
 
 
 static string shaderHeader(string header, int maxLights, bool hasTexture){

